p4app-exercises/basic_tunnel.p4app/README.md

```diff
@@ -28,20 +28,21 @@ rule.
 
 For this exercise, we have already added the necessary static control plane
 entries. As part of bringing up the Mininet instance, the `make run` command
-will install packet-processing rules in the tables of each switch. These are
-defined in the `sX-runtime.json` files, where `X` corresponds to the switch
-number.
+will install packet-processing rules in the tables of each switch.
+`make run` causes the program `main.py` to be run inside the
+new docker container. The control plane rules are installed near the
+end of `main.py`, by calls to the method `insertTableEntry`.
 
 Since the control plane tries to access the `myTunnel_exact` table, and that
 table does not yet exist, the `make run` command will not work with the starter
 code.
 
-**Important:** We use P4Runtime to install the control plane rules. The content
-of files `sX-runtime.json` refer to specific names of tables, keys, and
+**Important:** We use P4Runtime to install the control plane rules. The
+`insertTableEntry` method calls refer to specific names of tables, keys, and
 actions, as defined in the P4Info file produced by the compiler (look for the
-file `build/basic.p4info` after executing `make run`). Any changes in the P4
+file `/tmp/p4app-logs/basic_tunnel.p4info` after executing `make run`). Any changes in the P4
 program that add or rename tables, keys, or actions will need to be reflected
-in these `sX-runtime.json` files.
+in these `insertTableEntry` method calls in `main.py`.
```

Comment:
This paragraph doesn't seem accurate unless I'm missing something.
main.py executes mycontroller.py (towards the end of the file) as a child process, when `make run` is used. But the `insertTableEntry` calls are in mycontroller.py and can be edited there. They are not in main.py.

Reply:
Oops my bad I was looking at the wrong exercise...